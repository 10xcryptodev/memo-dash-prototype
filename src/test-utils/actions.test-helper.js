import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'

const verifyAction = (type, payload, actionCreator) => {
  const expectedAction = {
    type,
    payload
  }
  expect(actionCreator(payload)).toEqual(expectedAction)
}

const mockStore = configureMockStore([thunk])

const mockStoreAndDispatch = async (state, actionCreator) => {
  const store = mockStore(state)
  await store.dispatch(actionCreator)
  return store.getActions()
}

const findAction = (actions, type) => actions.find(action => action.type === type)

const getAction = (actions, type) => {
  const action = findAction(actions, type)
  if (action) return Promise.resolve(action)

  return new Promise(resolve => {
    store.subscribe(() => {
      const action = findAction(store, type)
      if (action) resolve(action)
    })
  })
}

export { verifyAction, mockStoreAndDispatch, getAction }
