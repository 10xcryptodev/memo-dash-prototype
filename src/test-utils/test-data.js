export const testUsers = {
  alice: {
    username: 'alice',
    profile: {
      avatarUrl: 'https://s3.amazonaws.com/uifaces/faces/twitter/madcampos/128.jpg',
      bio: 'Quod architecto molestias et.',
      followersCount: 1,
      followingCount: 1,
      likesCount: 2,
      username: 'alice'
    },
    userId: 'f0fd479ba875bf2232a4043dbefff24e2af2ca5ccd50b3ba9e91673707336726',
    likes: [
      {
        idx: 5,
        relation: {
          username: 'bob',
          userId: 'c6e5e284054bcbcd2cddea1ec36579e1fdf8e788d8a46d0351e276b4d0bb297c',
          index: 1
        }
      },
      {
        idx: 6,
        relation: {
          username: 'bob',
          userId: 'c6e5e284054bcbcd2cddea1ec36579e1fdf8e788d8a46d0351e276b4d0bb297c',
          index: 2
        }
      }
    ],
    memoIds: ['[alice][1]', '[alice][2]'],
    followers: ['bob'],
    following: ['bob']
  },
  bob: {
    username: 'bob',
    profile: {
      avatarUrl: 'https://s3.amazonaws.com/uifaces/faces/twitter/madcampos/129.jpg',
      bio: 'Id qui nisi asperiores vel quo.',
      followersCount: 1,
      followingCount: 1,
      likesCount: 1,
      username: 'bob'
    },
    userId: 'c6e5e284054bcbcd2cddea1ec36579e1fdf8e788d8a46d0351e276b4d0bb297c',
    likes: [
      {
        idx: 7,
        relation: {
          username: 'alice',
          userId: 'f0fd479ba875bf2232a4043dbefff24e2af2ca5ccd50b3ba9e91673707336726',
          index: 1
        }
      },
      {
        idx: 8,
        relation: {
          username: 'alice',
          userId: 'f0fd479ba875bf2232a4043dbefff24e2af2ca5ccd50b3ba9e91673707336726',
          index: 2
        }
      }
    ],
    memoIds: ['[bob][1]', '[bob][2]'],
    followers: ['alice'],
    following: ['alice']
  }
}

export const testMemos = {
  '[bob][1]': {
    avatarUrl: 'https://s3.amazonaws.com/uifaces/faces/twitter/rangafangs/128.jpg',
    idx: 1,
    memoDatetime: '2018-07-26T16:45:08.218Z',
    memoLikesCount: 0,
    memoRepliesCount: 0,
    memoText: 'Qui magnam maiores laboriosam animi sit aut.',
    memoTipTotal: 0,
    username: 'bob'
  },
  '[bob][2]': {
    avatarUrl: 'https://s3.amazonaws.com/uifaces/faces/twitter/rangafangs/128.jpg',
    idx: 2,
    memoDatetime: '2018-07-26T16:45:10.954Z',
    memoLikesCount: 0,
    memoRepliesCount: 0,
    memoText: 'Quia est odio nemo deserunt.',
    memoTipTotal: 0,
    username: 'bob'
  },
  '[alice][1]': {
    avatarUrl: 'https://s3.amazonaws.com/uifaces/faces/twitter/madcampos/128.jpg',
    idx: 1,
    memoDatetime: '2018-07-26T16:44:56.000Z',
    memoLikesCount: 1,
    memoRepliesCount: 0,
    memoText: 'Suscipit vel nobis.',
    memoTipTotal: 0,
    username: 'alice'
  },
  '[alice][2]': {
    avatarUrl: 'https://s3.amazonaws.com/uifaces/faces/twitter/madcampos/128.jpg',
    idx: 2,
    memoDatetime: '2018-07-26T16:44:58.620Z',
    memoLikesCount: 0,
    memoRepliesCount: 0,
    memoText: 'Praesentium libero est repudiandae et repudiandae.',
    memoTipTotal: 0,
    username: 'alice'
  }
}
