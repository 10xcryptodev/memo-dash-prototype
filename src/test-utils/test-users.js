export default [
  {
    username: 'alice',
    profile: {
      avatarUrl: 'https://s3.amazonaws.com/uifaces/faces/twitter/madcampos/128.jpg',
      bio: 'Quod architecto molestias et.',
      followersCount: 1,
      followingCount: 1,
      likesCount: 1,
      username: 'alice'
    },
    ownLikes: [{ idx: 1, relation: { index: 3 } }, { idx: 2, relation: { index: 4 } }],
    memos: [
      {
        avatarUrl: 'https://s3.amazonaws.com/uifaces/faces/twitter/chaabane_wail/128.jpg',
        idx: 1,
        memoDatetime: '2018-07-26T16:44:56.000Z',
        memoLikesCount: 1,
        memoRepliesCount: 0,
        memoText: 'Suscipit vel nobis.',
        memoTipTotal: 0,
        username: 'alice'
      },
      {
        avatarUrl: 'https://s3.amazonaws.com/uifaces/faces/twitter/chaabane_wail/128.jpg',
        idx: 2,
        memoDatetime: '2018-07-26T16:44:58.620Z',
        memoLikesCount: 0,
        memoRepliesCount: 0,
        memoText: 'Praesentium libero est repudiandae et repudiandae.',
        memoTipTotal: 0,
        username: 'alice'
      }
    ]
  },
  {
    username: 'bob',
    profile: {
      avatarUrl: 'https://s3.amazonaws.com/uifaces/faces/twitter/madcampos/129.jpg',
      bio: 'Id qui nisi asperiores vel quo.',
      followersCount: 1,
      followingCount: 1,
      likesCount: 1,
      username: 'bob'
    },
    ownLikes: [{ idx: 1, relation: { index: 3 } }, { idx: 2, relation: { index: 4 } }],
    memos: [
      {
        avatarUrl: 'https://s3.amazonaws.com/uifaces/faces/twitter/rangafangs/128.jpg',
        idx: 1,
        memoDatetime: '2018-07-26T16:45:08.218Z',
        memoLikesCount: 0,
        memoRepliesCount: 0,
        memoText: 'Qui magnam maiores laboriosam animi sit aut.',
        memoTipTotal: 0,
        username: 'bob'
      },
      {
        avatarUrl: 'https://s3.amazonaws.com/uifaces/faces/twitter/rangafangs/128.jpg',
        idx: 2,
        memoDatetime: '2018-07-26T16:45:10.954Z',
        memoLikesCount: 0,
        memoRepliesCount: 0,
        memoText: 'Quia est odio nemo deserunt.',
        memoTipTotal: 0,
        username: 'bob'
      }
    ]
  }
]
