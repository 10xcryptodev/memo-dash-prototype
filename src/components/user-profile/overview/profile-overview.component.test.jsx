import React from 'react'
import { shallow } from 'enzyme'
import { Card } from 'semantic-ui-react'

import { testUsers } from '../../../test-utils/test-data'
import ProfileOverviewComponent from './profile-overview.component'

describe('<ProfileOverviewComponent />', () => {
  let wrapper
  const alice = testUsers['alice']

  beforeEach(() => {
    const div = document.createElement('div')
    document.body.appendChild(div)
  })

  describe('should render', () => {
    it('without crashing', () => {
      wrapper = shallow(<ProfileOverviewComponent userProfile={alice.profile} />)
      expect(wrapper).toMatchSnapshot()
    })

    it('user followed by me', () => {
      wrapper = shallow(<ProfileOverviewComponent userProfile={alice.profile} following={true} />)
      expect(wrapper).toMatchSnapshot()
    })

    it('user not followed by me', () => {
      wrapper = shallow(<ProfileOverviewComponent userProfile={alice.profile} following={false} />)
      expect(wrapper).toMatchSnapshot()
    })

    it('my user', () => {
      wrapper = shallow(
        <ProfileOverviewComponent userProfile={alice.profile} isProfileOfCurrentUser={true} />
      )
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('interaction', () => {
    const fakeClickEvent = { stopPropagation() {} }
    let onGoToProfileClickedSpy

    beforeEach(() => {
      onGoToProfileClickedSpy = jest.fn()
      wrapper = shallow(
        <ProfileOverviewComponent
          userProfile={alice.profile}
          onGoToProfileClicked={onGoToProfileClickedSpy}
        />
      )
    })

    it('should call onLikeMemoClicked when clicked on like', () => {
      wrapper.find(Card).simulate('click', fakeClickEvent)
      expect(onGoToProfileClickedSpy).toHaveBeenCalled()
    })
  })
})
