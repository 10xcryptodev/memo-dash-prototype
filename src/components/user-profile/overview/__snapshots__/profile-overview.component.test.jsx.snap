// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileOverviewComponent /> should render my user 1`] = `
<Card
  as="div"
  link={true}
  onClick={[Function]}
>
  <Image
    as="img"
    fluid={true}
    src="https://s3.amazonaws.com/uifaces/faces/twitter/madcampos/128.jpg"
    ui={true}
  />
  <CardContent>
    <CardHeader>
      alice
      <Connect(FollowButtonComponent)
        userProfile={
          Object {
            "avatarUrl": "https://s3.amazonaws.com/uifaces/faces/twitter/madcampos/128.jpg",
            "bio": "Quod architecto molestias et.",
            "followersCount": 1,
            "followingCount": 1,
            "likesCount": 1,
            "username": "alice",
          }
        }
      />
    </CardHeader>
    <CardDescription>
      Quod architecto molestias et.
    </CardDescription>
  </CardContent>
  <CardContent
    extra={true}
  >
    <Label>
      Followers
      <LabelDetail>
        1
      </LabelDetail>
    </Label>
    <Label>
      Following
      <LabelDetail>
        1
      </LabelDetail>
    </Label>
  </CardContent>
</Card>
`;

exports[`<ProfileOverviewComponent /> should render user followed by me 1`] = `
<Card
  as="div"
  link={true}
  onClick={[Function]}
>
  <Image
    as="img"
    fluid={true}
    src="https://s3.amazonaws.com/uifaces/faces/twitter/madcampos/128.jpg"
    ui={true}
  />
  <CardContent>
    <CardHeader>
      alice
      <Connect(FollowButtonComponent)
        userProfile={
          Object {
            "avatarUrl": "https://s3.amazonaws.com/uifaces/faces/twitter/madcampos/128.jpg",
            "bio": "Quod architecto molestias et.",
            "followersCount": 1,
            "followingCount": 1,
            "likesCount": 1,
            "username": "alice",
          }
        }
      />
    </CardHeader>
    <CardDescription>
      Quod architecto molestias et.
    </CardDescription>
  </CardContent>
  <CardContent
    extra={true}
  >
    <Label>
      Followers
      <LabelDetail>
        1
      </LabelDetail>
    </Label>
    <Label>
      Following
      <LabelDetail>
        1
      </LabelDetail>
    </Label>
  </CardContent>
</Card>
`;

exports[`<ProfileOverviewComponent /> should render user not followed by me 1`] = `
<Card
  as="div"
  link={true}
  onClick={[Function]}
>
  <Image
    as="img"
    fluid={true}
    src="https://s3.amazonaws.com/uifaces/faces/twitter/madcampos/128.jpg"
    ui={true}
  />
  <CardContent>
    <CardHeader>
      alice
      <Connect(FollowButtonComponent)
        userProfile={
          Object {
            "avatarUrl": "https://s3.amazonaws.com/uifaces/faces/twitter/madcampos/128.jpg",
            "bio": "Quod architecto molestias et.",
            "followersCount": 1,
            "followingCount": 1,
            "likesCount": 1,
            "username": "alice",
          }
        }
      />
    </CardHeader>
    <CardDescription>
      Quod architecto molestias et.
    </CardDescription>
  </CardContent>
  <CardContent
    extra={true}
  >
    <Label>
      Followers
      <LabelDetail>
        1
      </LabelDetail>
    </Label>
    <Label>
      Following
      <LabelDetail>
        1
      </LabelDetail>
    </Label>
  </CardContent>
</Card>
`;

exports[`<ProfileOverviewComponent /> should render without crashing 1`] = `
<Card
  as="div"
  link={true}
  onClick={[Function]}
>
  <Image
    as="img"
    fluid={true}
    src="https://s3.amazonaws.com/uifaces/faces/twitter/madcampos/128.jpg"
    ui={true}
  />
  <CardContent>
    <CardHeader>
      alice
      <Connect(FollowButtonComponent)
        userProfile={
          Object {
            "avatarUrl": "https://s3.amazonaws.com/uifaces/faces/twitter/madcampos/128.jpg",
            "bio": "Quod architecto molestias et.",
            "followersCount": 1,
            "followingCount": 1,
            "likesCount": 1,
            "username": "alice",
          }
        }
      />
    </CardHeader>
    <CardDescription>
      Quod architecto molestias et.
    </CardDescription>
  </CardContent>
  <CardContent
    extra={true}
  >
    <Label>
      Followers
      <LabelDetail>
        1
      </LabelDetail>
    </Label>
    <Label>
      Following
      <LabelDetail>
        1
      </LabelDetail>
    </Label>
  </CardContent>
</Card>
`;
